jormungandr-watchdog/src/service/stats.rs

```diff
@@ -1,53 +1,62 @@
+use atomig::Atomic;
+use std::sync::atomic::{AtomicU64, Ordering};
+
 /// keep some stats based on [Welford's online algorithm]
 ///
 /// [Welford's online algorithm]: https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_online_algorithm
 pub struct Stats {
-    count: u64,
-    mean: f64,
-    m2: f64,
+    count: AtomicU64,
+    mean: Atomic<f64>,
+    m2: Atomic<f64>,
 }
 
 impl Stats {
     pub fn new() -> Self {
         Self {
-            count: 0,
-            mean: 0.0,
-            m2: 0.0,
+            count: AtomicU64::new(0),
+            mean: Atomic::new(0.0),
+            m2: Atomic::new(0.0),
         }
     }
 
-    pub fn push(&mut self, entry: f64) {
-        self.count += 1;
-
-        if self.count == 1 {
-            self.mean = entry;
-            self.m2 = 0.0;
+    pub fn push(&self, entry: f64) {
+        self.count.fetch_add(1, Ordering::Relaxed);
+        let self_count = self.count.load(Ordering::Relaxed);
+        if self_count == 1 {
+            self.mean.store(entry, Ordering::Relaxed);
+            self.m2.store(0.0, Ordering::Relaxed);
         } else {
-            let delta = entry - self.mean;
-            self.mean += delta / self.count as f64;
-            let delta2 = entry - self.mean;
-            self.m2 += delta * delta2;
+            let delta = entry - self.mean.load(Ordering::Relaxed);
+            self.mean.store(
+                self.mean.load(Ordering::Acquire) + delta / self_count as f64,
+                Ordering::Release,
+            );
+            let delta2 = entry - self.mean.load(Ordering::Relaxed);
+            self.m2.store(
+                self.m2.load(Ordering::Acquire) + delta * delta2,
+                Ordering::Release,
+            )
```

Comment:
too many breaking point... the order of the operation mater. Don't use atomic. use a `std::sync::Mutex` instead here